src/tools/scripts/update_timestamps_versions.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+######################################
+## Custom Web Almanac script        ##
+######################################
+#
+# This script updates CSS and JS reference to new versions numbers
+# when they are included in a commit. This is used for cahce busting.
+# 
+# It also updates timestamps for any .md and .html files changed
+#
+# This script is run in a GitHub Action and should not need to be run manually
+#
+
+# echo commands to screen for debugging
+set -x
+
+if [ "$#" -eq 1 ]; then
+COMMIT_SHA=$1
+fi
+
+if [ "{COMMIT_SHA}" = "" ]; then
+    echo "Must supply commit sha"
+    exit 1
+fi
+ 
+
+# This script must be run from src directory
+if [ -d "src" ]; then
+  cd src
+fi
+
+# You have to do some funky stuff for the version of sed on MacOS
+# https://stackoverflow.com/questions/5694228/sed-in-place-flag-that-works-both-on-mac-bsd-and-linux
+# Will mostly be running through GitHub Action but handy to have it working for
+# local development and debugging of this script.
+# Default case for Linux sed, just use "-i -r"
+SED_FLAGS=(-i -r)
+if [ $(uname) = 'Darwin' ]; then
+  echo "Running MacOS"
+  SED_FLAGS=(-i "" -E)
+fi
+
+NEW_LONG_DATE=`date +%Y%m%d%H%M%S`
+NEW_SHORT_DATE=`date '+%Y-%m-%d'`
```

Comment:
This might fall under the category is nitpicking, but I feel like these two `date` commands are creating two sources of truth for the same data. This will usually not be an issue, but theoretically there is a possibility that when the day is about to change, if in that very moment these commands run, their values might be different. Additionally, by default, `date` uses the locale of the system, but it will be more consistent to use UTC time in situations like these.

```suggestion
NOW=`date -u`
NEW_LONG_DATE=`date -d "$NOW" '+%Y%m%d%H%M%S'`
NEW_SHORT_DATE=`date -d "$NOW" '+%Y-%m-%d'`
```

Reply:
That's VERY nitpicky - and it doesn't work on MacOS:

```sh
++ date -u
+ NOW='Thu  6 Aug 2020 19:52:27 UTC'
++ date -d 'Thu  6 Aug 2020 19:52:27 UTC' +%Y%m%d%H%M%S
usage: date [-jnRu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ... 
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
+ NEW_LONG_DATE=
++ date -d 'Thu  6 Aug 2020 19:52:27 UTC' +%Y-%m-%d
usage: date [-jnRu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ... 
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
+ NEW_SHORT_DATE=
+ echo

+ echo
```

And also won't that set the date to that time? Not really want my script changing my system clock, even if it is to the same time as just before?

Ultimately these dates are only used for the sitemap and the pages meta data. It's not critical they are accurate to the  smallest nano-second. So will leave as it is.

Reply:
Did take you advice on UTC though - thanks!

Reply:
> That's VERY nitpicky

:laughing:

> and it doesn't work on MacOS:

Oh no, yet another incompatibility between Linux and macOS. In Linux `-d/--date` is the argument to supply custom datetime instead of using the default now. In macOS this same input is mapped to DST setting. It looks like [it is possible to supply a custom datetime in macOS as well](https://stackoverflow.com/questions/30810599/date-d-not-working-on-mac), but the signature of the command will be different.

> And also won't that set the date to that time?

In Linux the purpose of that flag is different, it does not have any side-effects.

> It's not critical they are accurate to the smallest nano-second. So will leave as it is.

Fair enough!

Reply:
OK changed it to this:

```sh
NEW_LONG_DATE=`date -u +%Y%m%d%H%M%S`
NEW_SHORT_DATE=${NEW_LONG_DATE:0:4}-${NEW_LONG_DATE:4:2}-${NEW_LONG_DATE:6:2}
```